tests/cucumber/steps/steps.js

```diff
@@ -5419,6 +5426,29 @@ module.exports = function getSteps(options) {
     }
   );
 
+  Then(
+    'the current application should have the following boxes {string}.',
+    async function (boxNames) {
+      let boxes;
+      if (boxNames !== '') {
+        boxes = splitAndProcessAppArgs(boxNames);
+      } else {
+        boxes = [];
+      }
+
+      const resp = await this.v2Client
+        .getApplicationBoxes(this.currentApplicationIndex)
+        .do();
+      const actualBoxes = new Set();
+      for (let i = 0; i < resp.boxes.length; i++) {
+        const boxName = resp.boxes[i].name;
+        actualBoxes.add(Buffer.from(boxName, 'base64'));
+      }
+      const expectedBoxes = new Set(boxes.map((name) => Buffer.from(name)));
+      assert.deepStrictEqual(expectedBoxes, actualBoxes);
```

Comment:
@algochoi Leaving a visual reminder from our verbal to additionally assert returned array lengths match.

Reply:
done, thanks!